tests/ui/formatting.rs

```diff
@@ -108,6 +108,25 @@ fn main() {
     a = *&191;
     b = !false;
 
+    // weird binary operator formatting:
```

Comment:
Could you move these tests to a new test file named after the lint? (You'll also have to update the stderr aftewards)

Reply:
Sure, done!